modules/cli/src/main/scala/scala/cli/commands/util/SharedOptionsUtil.scala

```diff
@@ -169,7 +170,12 @@ object SharedOptionsUtil extends CommandHelpers {
           addScalaLibrary = scalaLibrary.orElse(java.map(!_)),
           generateSemanticDbs = semanticDb,
           scalacOptions = ShadowingSeq.from(
-            scalac.scalacOption
+            scalac
+              .scalacOptionKeyFilter(
+                // -O -classpath should be redirected as --extra-jars instead
+                // similarly for -O -d, it should be redirected as --compilation-output
+                key => key != "-classpath" && key != "-d"
```

Comment:
It seems that these functions should be omitted everywhere, maybe you should add this filter [here](https://github.com/VirtusLab/scala-cli/blob/2e0b4b4c6816dba6004999db2be36e31f57c89d1/modules/options/src/main/scala/scala/build/options/ScalacOpt.scala#L7).

Reply:
still, the values for these keys have to be extracted first, so that they are redirected to the appropriate Scala CLI options.
I added a function in `ScalacOptionsUtil` for filtering them out, let me know if it's better now.

Reply:
It's good enough for me, thanks.